### PR TITLE
Let users edit their posts

### PR DESCRIPTION
**Pull request by browning**
Hey, just a change to let users edit their own posts. Also lets the admin edit any post.
Let me know if you think we should do it differently.
Thanks!

### CODE REVIEW

File: app/controllers/forem/posts_controller.rb [hunk 1]
```diff
@@ -30,6 +30,22 @@ def create
...
+    def edit
+      @topic = Topic.find(params[:topic_id])
+      @post = Post.find(params[:id])
+    end
+
+    def update
+      @post = Post.find(params[:id])
+      if @post.owner_or_admin?(forem_user) and @post.update_attributes(params[:post])
+        flash[:notice] = t("forem.post.edited")
+        redirect_to [@topic.forum, @topic]
```
Comment (parndt) [inline app/controllers/forem/posts_controller.rb]: Can we use this here instead?

``` ruby
redirect_to [@topic.forum, @topic], :notice => t('edited', :scope => 'forem.post')
```

That sorta redirect code is built into Rails now

Comment (radar) [inline app/controllers/forem/posts_controller.rb]: I agree, probably better.

**Comment by radar** [comment]
Hey, thanks for the terrific patch!

There's a comment [https://github.com/radar/forem/issues/53](here) that asks if we could have permissions determining if users could edit posts. Do you reckon you could write a little more into this patch to support that?

That'd be awesome! :)

**Comment by browning** [comment]
Yep, I will take a look at adding permissions tonight

**Comment by browning** [comment]
Alright, I think this is good to go now

**Comment by radar** [comment]
Neat work! Hooray for awesome contributors!

I just made a small change in @01c3a39 to allow people to customise where users can edit posts, so that some users can edit posts in some forums but not others if they feel like it.

**Pull request merged**